java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -556,12 +556,12 @@ public void restore() {
   private native void nativeRestore();
 
   /**
-   * Creates a new index on the dataset. Only vector indexes are supported.
+   * Creates a new index on the dataset.
    *
    * @param columns the columns to index from
    * @param indexType the index type
    * @param name the name of the created index
-   * @param params index params
+   * @param params index params, can be null for non-vector index types
```

Comment:
I think it will not be null? since user will use `IndexParams.create()` it will always be a non-null value. It does not handle null in the JNI code but I think it's probably better to enforce it, because we explicitly use `Optional` for nullable objects in methods.